owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt

```diff
@@ -250,31 +251,58 @@ fun Activity.hideSoftKeyboard() {
 fun Activity.checkPasscodeEnforced(securityEnforced: ISecurityEnforced) {
     val sharedPreferencesProvider = SharedPreferencesProviderImpl(this)
 
-    val lockEnforced = this.resources.getBoolean(R.bool.lock_enforced)
+    val lockEnforced: Int = this.resources.getInteger(R.integer.lock_enforced)
     val passcodeConfigured = sharedPreferencesProvider.getBoolean(PassCodeActivity.PREFERENCE_SET_PASSCODE, false)
     val patternConfigured = sharedPreferencesProvider.getBoolean(PatternActivity.PREFERENCE_SET_PATTERN, false)
 
-    if (lockEnforced && !passcodeConfigured && !patternConfigured) {
-
-        val options = arrayOf(getString(R.string.security_enforced_first_option), getString(R.string.security_enforced_second_option))
-        var optionSelected = 0
-
-        AlertDialog.Builder(this).apply {
-            setCancelable(false)
-            setTitle(getString(R.string.security_enforced_title))
-            setSingleChoiceItems(options, optionSelected) { _, which -> optionSelected = which }
-            setPositiveButton(android.R.string.ok) { dialog, _ ->
-                when (optionSelected) {
-                    0 -> securityEnforced.optionLockSelected(LockType.PASSCODE)
-                    1 -> securityEnforced.optionLockSelected(LockType.PATTERN)
-                }
-                dialog.dismiss()
+    when (lockEnforced) {
+        LockEnforcedType.DIALOG_OPTION.ordinal -> {
```

Comment:
I thought that working with ordinals was the right way because I saw them in several open source projects. But I agree that pasring the values is more understandable.